mmdet/models/detectors/single_stage.py

```diff
@@ -101,9 +105,14 @@ def simple_test(self, img, img_metas, rescale=False):
         feat = self.extract_feat(img)
         results_list = self.bbox_head.simple_test(
             feat, img_metas, rescale=rescale)
+        if to_bbox_results:
+            return self.post_process(results_list)
+        return results_list
+
+    def post_process(self, results):
         bbox_results = [
             bbox2result(det_bboxes, det_labels, self.bbox_head.num_classes)
-            for det_bboxes, det_labels in results_list
+            for det_bboxes, det_labels in results
         ]
```

Comment:
It's unnecessary to put such a simple line into a method.

```
        if to_bbox_results:
            bbox_results = [
                bbox2result(det_bboxes, det_labels, self.bbox_head.num_classes)
                for det_bboxes, det_labels in results_list
                for det_bboxes, det_labels in results
            ]
            return bbox_results
        return results_list
```

Reply:
simple_test will run on IPU, but bbox2result calculates result by numpy array, it cannot be traced, so this function should be called separately while in IPU mode.

Reply:
Okay, better to rename this method to `convert_results_format`. The name "post_process" could be confused with the post-process of the algorithm.